pyam/core.py

```diff
@@ -1850,6 +1850,37 @@ def divide(self, a, b, name, axis="variable", append=False):
         else:
             return IamDataFrame(_value, meta=self.meta)
 
+    def apply(self, name, components, func, axis="variable", append=False):
```

Comment:
This use of `components` could become confusing with more elaborate custom functions - I suggest to use `kwds` instead of `components` to follow the pandas apply function (see [here](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.apply.html)), so that the function could be used in this manner:

```python
def custom_calculation(a, b, c):
    return a + b / c

df.apply(custom_calculation, name="new variable", a="Primary Energy", b="Secondary Energy", c="Price")
```